### ISSUE TITLE
Warn about yen signs (¥) or won signs (₩) followed by an escape character

### ISSUE THREAD

**Issue by xfix** [rust-lang/rust-clippy#7964]
### What it does

Warns about U+00A5, U+FFE5, U+20A9 or U+FFE6 followed by an escape character (`x`, `n`, `r`, `t`, `u`) or itself.

### Categories (optional)

- Kind: suspicious

Doing so suggests the intent to actually use U+005C (\\) instead. When using Windows in Japanese locale, U+005C is rendered as a Yen sign. When using Windows in Korean locale, U+005C is rendered as a won sign.

### Drawbacks

None.

### Example

```rust
print!("Hello¥n");
```

Could be written as:

```rust
print!("Hello\u{A5}n");
```

or what probably intended:

```rust
print!("Hello\n");
```

**Comment by xfix** [comment]
Examples of violations:

- https://github.com/Yamato-Security/hayabusa/blob/b97b22403eb4ef4633b7e5a32b499051ae7e395e/src/detections/rule/count.rs#L850
- https://github.com/ktanaka101/monkey.rs/blob/bc000a6db7fbf49d26b9ba6390a8aa2f66e9f0a6/src/compiler.rs#L1612
